R/corpus-methods-quanteda.R

```diff
@@ -120,10 +119,9 @@ as.corpus.default <- function(x) {
 #' @export
 #' @method as.corpus corpus
 as.corpus.corpus <- function(x) {
-    
     if (is.character(x)) {
         attr(x, "docvars") <- upgrade_docvars(attr(x, "docvars"))
-    } else {
+    } else if (is.null(meta(x, type = "system")[["package-version"]])) {
         x <- upgrade_corpus(x)
```

Comment:
You should change the `upgrade_corpus()` a bit, because there are lines like:
`attr(result, "meta")$created <- as.POSIXlt(Sys.time())`